guide/17-working-with-knowledge-graphs/part2_search_query_knowledge_graph.ipynb

```diff
@@ -112,8 +113,110 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# using bind parameters in queries\n",
+    "\n",
+    "# list example\n",
+    "query_list = ['Megan', 'Emma', 'Cameron', 'Noah']\n",
+    "results = knowledge_graph.query_streaming(\"MATCH (p:Person) WHERE p.name IN $list RETURN p\", bind_param={\"list\": query_list})\n",
     "\n",
+    "# anonymous object example\n",
+    "query_obj = {\"props\": {\"name\": \"Megan\"}, \"list\": ['Emma', 'Cameron', 'Noah']}\n",
+    "results = knowledge_graph.query_streaming(\"MATCH (n:Person)-[:FriendsWith]-(e:Person) WHERE n.name = $object.props.name AND e.name in $object.list RETURN n, e\", bind_param={\"object\": query_obj})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The output of these queries are a generator, so they need to be handled slightly different from the regular query output."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# handling results - get all results\n",
+    "for result in list(results):\n",
+    "    # do something with each result\n",
+    "    print(result)\n",
+    "\n",
+    "# handling results - get results one at a time\n",
+    "next(results)\n",
+    "\n",
+    "# or loop through all results using next\n",
+    "while True:\n",
+    "    try:\n",
+    "        # do something with each result\n",
+    "        print(next(results))\n",
+    "    except StopIteration:\n",
+    "        break"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# including provenance in query results\n",
+    "results = knowledge_graph.query_streaming(\"MATCH (n:Provenance) RETURN n LIMIT 1\", include_provenance=True)\n",
+    "list(results)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The result of this query would look similar to:\n",
+    "```\n",
+    "[\n",
+    "    [\n",
+    "        {'_objectType': 'entity',\n",
+    "        '_typeName': 'Provenance',\n",
+    "        '_id': UUID('1794b6b2-4d91-48ad-b51f-5dfb80e58c01'),\n",
+    "        '_properties': {'instanceID': UUID('3e16d8fe-7f68-45ef-805a-a54d78995411'),\n",
+    "            'propertyName': 'name',\n",
+    "            'sourceType': 'String',\n",
+    "            'typeName': 'Document',\n",
+    "            'globalid': UUID('1794b6b2-4d91-48ad-b51f-5dfb80e58c01'),\n",
+    "            'sourceName': 'MySourceName',\n",
+    "            'source': 'MySource',\n",
+    "            'objectid': 2\n",
+    "            }\n",
+    "        }\n",
+    "    ]\n",
+    "]\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```

Comment:
* Reads and looks great when built locally:
![image](https://github.com/Esri/arcgis-python-api/assets/1399179/d7878efe-af76-441f-902c-613f29d42aaf)